src/app/hooks/useGlean.ts

```diff
@@ -4,46 +4,116 @@
 
 "use client";
 
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
 import Glean from "@mozilla/glean/web";
-import * as pageEvents from "../../telemetry/generated/page";
+// @ts-expect-error We do not have type declarations for metrics.yaml, yet.
+import gleanMetrics from "../../telemetry/metrics.yaml";
+import {
+  SnakeToCamelCase,
+  convertSnakeToCamelCase,
+} from "../functions/universal/convertSnakeToCamelCase";
+
+const APP_ENV = process.env.NEXT_PUBLIC_APP_ENV || "";
+
+export type GleanExtraKeys = Record<string, string | boolean | number>;
+
+type GleanCategoryKey = keyof typeof gleanMetrics;
+
+type GleanEvent = {
+  [key: string]: {
+    type: string;
+    description: string;
+    bugs: string;
+    data_reviews: string;
+    data_sensitivity: string;
+    notification_emails: string;
+    expires: string;
+    extra_keys: GleanExtraKeys;
+    record: (params: GleanExtraKeys) => void;
+  };
+};
+
+type GleanCategories = {
+  [key: GleanCategoryKey]: GleanEvent;
+};
+
+type GleanMetrics = {
+  $schema: string;
+} & GleanCategories;
```

Comment:
All of the types in this file are not final and we might be able to get some from `@mozilla/glean`.